io.github.mhogomchungu.sirikali.json

```diff
@@ -0,0 +1,48 @@
+{
+  "app-id" : "io.github.mhogomchungu.sirikali",
+  "runtime" : "org.kde.Platform",
+  "runtime-version" : "6.7",
+  "sdk" : "org.kde.Sdk",
+  "command" : "io.github.mhogomchungu.sirikali",
+  "finish-args" : [
+    "--share=ipc",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--filesystem=xdg-documents",
+    "--share=network",
+    "--device=dri",
+    "--talk-name=org.kde.StatusNotifierWatcher",
+    "--talk-name=org.freedesktop.Flatpak"
+  ],
+  "modules" : [
+    {
+      "name" : "libfuse",
+      "buildsystem" : "autotools",
+      "config-opts" : [
+        "--disable-util"
+      ],
+      "sources" : [
+        {
+          "type" : "archive",
+          "url" : "https://github.com/libfuse/libfuse/releases/download/fuse-2.9.9/fuse-2.9.9.tar.gz",
+          "sha256" : "d0e69d5d608cc22ff4843791ad097f554dd32540ddc9bed7638cc6fea7c1b4b5"
+        }
+      ]
+    },
+    {
+      "name" : "io.github.mhogomchungu.sirikali",
+      "buildsystem" : "cmake-ninja",
+      "config-opts" : [
+        "-DCMAKE_BUILD_TYPE=release",
+        "-DBUILD_FLATPAK=TRUE"
```

Comment:
This is just a drive by comment. Doing this upstream makes no sense. Flatpak isn't a special target, you are supposed to install the same files in the same places with the same names always on Linux.

Reply:
Flatpak is a special platform and other supported platforms are linux, windows and macos.

In Linux platform :-

1. Executables are expected to be found in all sort of places.
2. Executables are called simply by their name.
3. Some Cryfs arguments are pass as environmental variables in a normal way.
4. Default home is "~".
5. "/proc/self/mountinfo" works as expected because it gets updated when volumes are mounted and unmounted.
6. Backends run as background tasks because their status is discovered by reading "/proc/self/mountinfo" file. This makes it possible for SiriKali to exit while they are running and it rediscovers them when it starts again.
7. All sorts of fuse based programs are supported including Gogcryptfs, CryFS and Securefs.
8. SiriKali executable is called "sirikali".

In flatpak platform:-

1. Executables are expected to be found in "~/.var/app/io.github.mhogomchungu.sirikali/data/SiriKali/bin/".
2. Executables are called through flatpak-spawn.
3. Some Cryfs arguments are pass as environmental variables through flatpal-spawn's --args argument.
4. Default home is "~/Documents".
5. My tests shows that "/proc/self/mountinfo" is static, it works as expected the first time the file is read and then subsequent reads return the same data that was read the first time. Because of this, flatpak version does not depend on this file.
6. Backend run as foreground tasks and this means SiriKali has to babysit them and can not quit while they are running.
7. Only Securefs, CryFs and Gocrypts are supported.
8. SiriKali executable is called "io.github.mhogomchungu.sirikali".

Reply:
> Executables are expected to be found in "~/.var/app/io.github.mhogomchungu.sirikali/data/SiriKali/bin/".

They go in `/app/bin` which is basically `$prefix/bin` same as everything else.

> Default home is "~/Documents".

No

> SiriKali executable is called "io.github.mhogomchungu.sirikali".

There is no need to name it like that.

Reply:
>> Executables are expected to be found in "~/.var/app/io.github.mhogomchungu.sirikali/data/SiriKali/bin/".

> They go in /app/bin which is basically $prefix/bin same as everything else.

Backends are not bundled with SiriKali and they are expects to be at "${QStandardPaths::AppDataLocation}/bin" which resolves to above location in flatpak.

>> Default home is "~/Documents".

> No

From the app's perspective, "Default home" is the default location of where new volumes and mount points are created. In flatpak, the path is  "${QStandardPaths::DocumentsLocation]". In Windows, the path is "${QStandardPaths::HomeLocation}/Desktop". In Linux, the path is "${QStandardPaths::HomeLocation}".

>> SiriKali executable is called "io.github.mhogomchungu.sirikali".

> There is no need to name it like that.

True, just though i should name it this was for consistency sake.

Reply:
> Backends are not bundled with SiriKali and they are expects to be at "${QStandardPaths::AppDataLocation}/bin" which resolves to above location in flatpak.

How are these backends supposed to be used? Should the user download them and place them or the app does it? If it is necesaary for the app to function, these should be in `$prefix/share/appname/`

Reply:
1. This program is a front end to other tools  like gocryptfs.
2. It downloads gocryptfs for example from its github webpage and installs  it to its *private* writable location.

More info about this feature is discussed [here](https://github.com/mhogomchungu/sirikali/issues/261)

Reply:
That is fine, it has no relevance to the binary and metadata installed for the program itself.